pkg/controller/druid/handler.go

```diff
@@ -879,6 +926,28 @@ func namespacedName(name, namespace string) *types.NamespacedName {
 	return &types.NamespacedName{Name: name, Namespace: namespace}
 }
 
+// IntPointer returns a pointer to int32.
+func IntPointer(i int32) *int32 {
+	return &i
+}
+
+// GetHPAReplicaCountOrDefault returns the count of pods in case hpa is mentioned.
+func GetHPAReplicaCountOrDefault(name types.NamespacedName, defaultReplicaCount int32) int32 {
+	var hpa autoscalev2beta1.HorizontalPodAutoscaler
+	var nodeSpec v1alpha1.DruidNodeSpec
+
+	err := &nodeSpec.Replicas
+	if err != nil {
+		return defaultReplicaCount
```

Comment:
other validator PR  *requires* `nodeSpec.Replicas` so , this case  will always be true once that  PR  is merged.